src/mme-app/utils/mmeS1MsgUtils.cpp

```diff
@@ -0,0 +1,111 @@
+/*
+* Copyright 2020-present, Infosys Ltd.
+*
+* SPDX-License-Identifier: Apache-2.0
+*/
+
+#include "utils/mmeS1MsgUtils.h"
+
+#include <controlBlock.h>
+#include <contextManager/dataBlocks.h>
+#include <contextManager/subsDataGroupManager.h>
+#include <log.h>
+#include <mme_app.h>
+#include <msgBuffer.h>
+#include <s1ap_structs.h>
+#include <utils/defaultMmeProcedureCtxt.h>
+#include "secUtils.h"
+using namespace mme;
+
+extern mme_config g_mme_cfg;
+
+void MmeS1MsgUtils::populateHoRequest(SM::ControlBlock& cb,
+        UEContext& ueCtxt,
+		S1HandoverProcedureContext& procCtxt,
+		struct handover_request_Q_msg& hoReq)
+{
+    SessionContext *sessionCtxt = ueCtxt.getSessionContext();
+    if (sessionCtxt == NULL)
+    {
+        log_msg(LOG_DEBUG,
+                " send_ho_request_to_target_enb: session ctxt is NULL \n");
+        return;
+    }
+    BearerContext *bearerCtxt = sessionCtxt->getBearerContext();
+    if (bearerCtxt == NULL)
+    {
+        log_msg(LOG_ERROR, "Failed to retrieve Bearer context for UE IDx %d\n",
+                cb.getCBIndex());
+        return;
+    }
+    hoReq.msg_type = handover_request;
+    hoReq.s1ap_mme_ue_id = ueCtxt.getContextID();
+    hoReq.enb_fd = procCtxt.getTargetEnbFd();
+
+    hoReq.handoverType = IntraLTE;
+    hoReq.cause.present = s1apCause_PR_radioNetwork;
+    hoReq.cause.choice.radioNetwork =
+            procCtxt.getS1apCause().s1apCause_m.choice.radioNetwork;
+
+    memcpy(&(hoReq.src_to_target_transparent_container),
+            &(procCtxt.getSrcToTargetTransContainer()),
+            sizeof(struct src_target_transparent_container));
+
+    hoReq.ue_aggrt_max_bit_rate.uEaggregateMaxBitRateDL =
+            (ueCtxt.getAmbr().ambr_m).max_requested_bw_dl;
+    hoReq.ue_aggrt_max_bit_rate.uEaggregateMaxBitRateUL =
+            (ueCtxt.getAmbr().ambr_m).max_requested_bw_ul;
+
+    hoReq.eRABSetupList.count = 1;
+    hoReq.eRABSetupList.eRABSetup[0].e_RAB_ID = 5;
+    hoReq.eRABSetupList.eRABSetup[0].gtp_teid =
+            bearerCtxt->getS1uSgwUserFteid().fteid_m.header.teid_gre;
+    hoReq.eRABSetupList.eRABSetup[0].transportLayerAddress =
+            bearerCtxt->getS1uSgwUserFteid().fteid_m.ip.ipv4.s_addr;
+
+    hoReq.gummei.mme_code = g_mme_cfg.mme_code;
+    hoReq.gummei.mme_grp_id = g_mme_cfg.mme_group_id;
+    memcpy(hoReq.gummei.plmn_id.idx, g_mme_cfg.plmn_id.idx, 3);
```

Comment:
your assumption is both end's having same plmn?

Reply:
No necessarily. Do you see any issue here?